core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/WorkflowWorker.scala

```diff
@@ -198,7 +196,7 @@ class WorkflowWorker(identifier: ActorVirtualIdentity, operator: IOperatorExecut
       }
     case Pause =>
       workerStateManager.confirmState(Running, Ready)
-      promiseManager.execute(ControlInvocation(null, WorkerPause()))
+      rpcServer.execute(ControlInvocation(RPCServer.RootCallWithoutReturn, WorkerPause()), null)
```

Comment:
I am not understanding. Shouldnt this be handled in controlInputPort? and why `RootCallWithoutReturn`. shouldn't controller send its VirtualIdentity?

Reply:
Again, for compatibility, I'm not changing the way the controller sends the control message so I have to write it this way.

Reply:
please add a todo so that we dont forget